bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/status/FindStatus.java

```diff
@@ -42,9 +42,7 @@ public boolean isInputValid() {
 
 	@Override
 	public boolean wasSuccessful() {
-		// Also report StatusCode.WRAPPED as unsuccessful, because it implicitly
-		// includes NO_MATCH, as before wrapping no further match was found
-		return false;
+		return messageCode != StatusCode.NO_MATCH;
```

Comment:
Shouldn't `READONLY` also be reported as not successful?
```suggestion
		return messageCode == StatusCode.WRAPPED;
```

Reply:
Yes, it should! Fixed in https://github.com/eclipse-platform/eclipse.platform.ui/commit/80f6de0c42e4cc17c3393545f38b500f1288ed52